ant-service-management/src/metric.rs

```diff
@@ -0,0 +1,216 @@
+        }
+    }
+    NetworkInfoMetrics::new(connected_peers, listeners)
+}
+
+#[derive(Debug, Clone)]
+pub struct MetricClient {
+    endpoint_port: String,
+}
+
+impl MetricClient {
+    pub fn new(endpoint_port: u16) -> Self {
+        Self {
+            endpoint_port: endpoint_port.to_string(),
+        }
+    }
+
+    pub async fn get_endpoint_metrics(&self, endpoint_name: &str) -> Result<prometheus_parse::Scrape, Box<dyn std::error::Error>> {
+            debug!(
+                "Attempting connection to collect {} metrics from {}...",
+                endpoint_name,
+                self.endpoint_port
+            );
+
+            let body = reqwest::get(&format!("http://localhost:{}/{endpoint_name}", self.endpoint_port))
+            .await?
+            .text()
+            .await?;
+            let lines: Vec<_> = body.lines().map(|s| Ok(s.to_owned())).collect();
+            let all_metrics = prometheus_parse::Scrape::parse(lines.into_iter())?;
+
+            Ok(all_metrics)
+    }
+
+    pub fn get_node_info_from_metadata_extended(&self ,scrape: &prometheus_parse::Scrape, node_info: &mut NodeInfoMetrics) {
+        for sample in scrape.samples.iter() {
+            for (key, value) in sample.labels.iter() {
+                match key.as_str() {
+                    "peer_id" => node_info.peer_id = value.parse().unwrap(),
+                    "pid" => node_info.pid = value.parse().unwrap(),
+                    "bin_version" => node_info.bin_version = value.parse().unwrap(),
+                    "root_dir" => node_info.root_dir =value.parse().unwrap(),
+                    "log_dir" => node_info.log_dir = value.parse().unwrap(),
+                    _ => {}
+                }
+            }
+        }
+    }
+
+    pub fn get_node_info_from_metrics(&self, scrape: &prometheus_parse::Scrape, node_info: &mut NodeInfoMetrics) {
+        for sample in scrape.samples.iter() {
+            if sample.metric == "ant_node_current_reward_wallet_balance" {
+                // Attos
+                match sample.value {
+                    prometheus_parse::Value::Counter(val)
+                    | prometheus_parse::Value::Gauge(val)
+                    | prometheus_parse::Value::Untyped(val) => {
+                        node_info.wallet_balance = val as u64;
+                    }
+                    _ => {}
+                }
+            } else if sample.metric == "ant_node_uptime" {
+                match sample.value {
+                    prometheus_parse::Value::Counter(val)
+                    | prometheus_parse::Value::Gauge(val)
+                    | prometheus_parse::Value::Untyped(val) => {
+                        node_info.uptime = Duration::new(val as u64, 0);
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+}
+
+#[async_trait]
+impl RpcActions for MetricClient {
+    async fn node_info(&self) -> Result<NodeInfo> {
```

Comment:
Regarding the `Action` here:
- We must rename `RpcActions` -> `MetricActions`
- remove the unused methods
- Also the `expect` has to be removed.

Reply:
Yeah, I was also puzzled about the use of `RpcActions`. We shouldn't have references to RPC any more.